### PR TITLE
Use custom classes to pass client signals parameters #2686

### PR DESCRIPTION
**Pull request by pfreixes**
## What do these changes do?

This will allow Aiohttp development add new parameters in the future without
break the signals signature.

## Are there changes in behavior for the user?

Yes

## Related issue number

#2686

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes

### CODE REVIEW

File: docs/tracing_reference.rst [hunk 1]
```diff
@@ -29,10 +29,10 @@ the request flow.
    .. attribute:: on_request_start
 
       Property that gives access to the signals that will be executed when a
-      request starts, based on the :class:`~signals.Signal` implementation.
+      request starts, based on the :class:`aiohttp.signals.Signal` implementation.
 
       The coroutines listening will receive as a param the ``session``,
-      ``trace_config_ctx``, ``method``, ``url`` and ``headers``.
+      ``trace_config_ctx`` and :class:`aiohttp.TraceRequestStartParams` params.
```
Comment (asvetlov) [inline docs/tracing_reference.rst]: Minor question.
Maybe ```The signal handler signature is ``async def on_request_start(session, context, params): ...`` where ``params`` is :class:`aiohttp.TraceRequestStartParams` instance``` for better readability?

The same for all other tracing signals.

**Comment by asvetlov** [comment]
Looks good in overall.

File: docs/tracing_reference.rst (hunk 1 above)
Comment (pfreixes) [reply docs/tracing_reference.rst]: Done!

**Comment by jettify** [comment]
I reworked aiozipkin (https://github.com/aio-libs/aiozipkin/pull/95) to use this API, I think new one is much better and easier to use! 👍

**Comment by asvetlov** [comment]
cool!

**Pull request merged**